s3sync.go

```diff
@@ -89,6 +86,22 @@ func (m *Manager) Sync(source, dest string) error {
 		return err
 	}
 
+	chJob := make(chan func())
+	var wg sync.WaitGroup
+	for i := 0; i < m.nJobs; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for job := range chJob {
+				job()
+			}
+		}()
+	}
+	defer func() {
+		close(chJob)
+		wg.Wait()
```

Comment:
The other wait groups in syncLocalToS3/syncS3ToLocal functions seem blocking until all jobs done. Do we really need this wait group?

Reply:
WaitGroups in `syncLocalToS3/syncS3ToLocal` wait until all jobs are finished. This WaitGroup is to ensure all job processor routines are stopped.
It's mainly for the cases that caller checks routine leak in a test using this library. (In most cases, there will be no problem without this WaitGroup.)

Reply:
> caller checks routine leak in a test using this library

Didn't think of that case... Thank you for the explanation. It makes sense to me now.